projects/rocfft/library/src/plan.cpp

```diff
@@ -2103,6 +2121,225 @@ void rocfft_plan_t::GlobalTransposeP2P(size_t                     elem_size,
+            if(outSubcommRank == -1)
+                continue;
+
+            auto intersection = inBrick.intersect(outBrick);
+            if(intersection.empty())
+                continue;
+            intersection.stride = intersection.contiguous_strides();
+
+            // pack if this rank owns input brick
+            if(inRank == my_global_rank)
+            {
+                auto pack_op = AddMultiPlanItem(
+                    transpose_brick(my_global_rank,
+                                    inBrick.location,
+                                    intersection.length(),
+                                    precision,
+                                    desc.inArrayType,
+                                    input[inBrickIdx],
+                                    intersection.offset_in_field(inBrick.stride)
+                                        - inBrick.offset_in_field(inBrick.stride),
+                                    inBrick.stride,
+                                    BufferPtr::temp(send_buf.data()),
+                                    send_offsets[outSubcommRank],
+                                    intersection.stride,
+                                    "pack brick for subcomm transpose"),
+                    inputAntecedents);
+                multiPlan[pack_op]->group = itemGroup;
+                multiPlan[pack_op]->description
+                    = "pack " + std::to_string(inBrickIdx) + " + " + std::to_string(outBrickIdx);
+                pack_ops.push_back(pack_op);
+            }
+
+            // unpack if this rank owns output brick
+            if(outRank == my_global_rank)
+            {
+                auto unpack_op = AddMultiPlanItem(
+                    transpose_brick(my_global_rank,
+                                    outBrick.location,
+                                    intersection.length(),
+                                    precision,
+                                    desc.inArrayType,
+                                    BufferPtr::temp(recv_buf.data()),
+                                    recv_offsets[inSubcommRank],
+                                    intersection.stride,
+                                    output[outBrickIdx],
+                                    intersection.offset_in_field(outBrick.stride)
+                                        - outBrick.offset_in_field(outBrick.stride),
+                                    outBrick.stride,
+                                    "unpack brick for subcomm transpose"),
+                    {});
+                multiPlan[unpack_op]->group = itemGroup;
+                multiPlan[unpack_op]->description
+                    = "unpack " + std::to_string(inBrickIdx) + " + " + std::to_string(outBrickIdx);
+                unpack_ops.push_back(unpack_op);
+            }
+        }
+    }
+
+    // check uniformity for alltoall within the subcomm
+    bool uniform_counts = std::all_of(send_counts.begin(),
```

Comment:
```suggestion
    const bool uniform_counts = std::all_of(send_counts.begin() + 1,
```
and elsewhere (it'd be worth defining a local lambda given its repeated use, imo).
\+ insisting on checking that `subcomm_size  > 0` above.

Reply:
This is fixed in the new commit.